src/services/ImageRepository.cpp

```diff
@@ -175,9 +188,17 @@ void ImageRepository::FetchImage(ImageType nType, const std::string& sName)
             return;
     }
 
-    // Prevent duplicate requests
-    if (!RAWeb::HTTPRequestExists(nRequestType, sName) && !RAWeb::HTTPResponseExists(nRequestType, sName))
-        RAWeb::CreateThreadedHTTPRequest(nRequestType, args, sName);
+    ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync([this, nRequestType, args, sFilename]()
+    {
+        std::string sResponse;
+        if (RAWeb::DoBlockingRequest(nRequestType, args, sResponse))
+            _WriteBufferToFile(sFilename, sResponse);
+
+        {
+            std::lock_guard<std::mutex> lock(m_oMutex);
+            m_vRequestedImages.erase(sFilename);
+        }
+    });
```

Comment:
Request the image and write it to disk all on the same background thread. No longer relies on the _RA_HandleHTTPResults method to write the image to disk.

Also co-locates the request with the response processing instead of making a roundabout loop through RA_Core.cpp